### DISCUSSION TITLE
what is the usage of `Default` keyword?

### DISCUSSION THREAD

**Discussion by vihsonic** [abharun/mini-chain#17] [Q&A]
https://github.com/abharun/mini-chain/blob/091623d67fca3e98a1531067d02275b3ab3ff15d/src/mini_chain/block.rs#L16
here `Default` is default trait in rust?
Why don't you create simple method within that struct? (e.g. `new()`)

**Comment by abharun** [comment]
In general, there's no big difference between them.
The best practice for defining the constructor is using `Default` trait for the struct.

`new()` provides more flexibility and fit for initiatives with user modifications. So when a struct variable is just constructed just once with default value at the beginning of the project, `Default` would be the practice. Otherwise, if you need to define several struct variables with user customized value, `new()` is better.

For instance, when you create a new `client` with Network entrypoints, you should define `new()` constructor that stores the received network entrypoints within the client struct that is required for broadcast triggered transactions.

On the other hand, when you create a chain `metadata` struct, it doesn't require any user modified configuration but just initial values, as you can just define `default()` constructor for initialize.
